Implemented Revelation Dance

lucfd opened the pull request:
~~- Added 'Typeless' type~~
~~- has neutral attacking & defensive matchups~~
~~- is assigned to Pokemon that have no types~~
  ~~-is removed from Pokemon that gain a type~~
- Created new attribute `MatchUserTypeAttr` that changes a move's typing to match the user's primary type.
- Implemented Revelation Dance.
  - Matches the user's primary type
  - Can become typeless
  - Changes to your active tera type (except for stellar, in which case it uses your previous type)
- Fixed Steam Engine activating on status moves.

Xavion3 commented:
This implementation is incomplete, Revelation Dance should be able to become typeless if the user is typeless somehow, but also Revelation Dance shouldn't change type if the user is tera stellar, and currently it looks like it would become stellar type.

lucfd commented:
Good catch on the stellar type behaviour. Revelation Dance will work with typeless once it's implemented (presumably it will be another type with neutral matchups across the board), but I guess now's as good a time as ever to add it

lucfd commented:
I believe so. I'm going off of this video, which is weirdly the only thing I can find on this edge case https://www.youtube.com/watch?v=sX23y80fYw8

Flashfyre commented:
There's a chance this will crash somewhere because there's no type image for it. I will need to test this.

lucfd commented:
In my testing it pulled a question mark image which I presume is a default case

Procyonae commented:
You could make stuggle use your new typeless type too so there's not two ways of doing it simultaneously

lucfd commented:
I've removed `Type.TYPELESS` completely and switched everything to the `Type.UNKNOWN` implementation agreed upon in Discord. Let me know if any other changes need to be made!

Xavion3 reviewed (commented):
This looks generally fine though, just those few small things.

lucfd commented on src/data/type.ts:
```diff
@@ -543,4 +543,4 @@ export function getTypeRgb(type: Type): [ integer, integer, integer ] {
   }
-}
\ No newline at end of file
+}
```
This diff is caused by the addition of an EOF newline character

lucfd commented:
Debug statement removed & test conditions have been simplified

chacolah commented:
Tested with fire Oricorio after rebasing current mainbranch, no issues, works as intended.

Xavion3 reviewed (commented):
Looks good now
